src/commands/init/index.ts

```diff
@@ -186,16 +186,24 @@ export class Init extends SwankyCommand<typeof Init> {
       },
     ];
 
+    this.configBuilder.defaultAccount = "alice";
+
     Object.keys(this.configBuilder.contracts!).forEach(async (contractName) => {
       await ensureDir(path.resolve(this.projectPath, "artifacts", contractName));
       await ensureDir(path.resolve(this.projectPath, "tests", contractName));
     });
 
     this.taskQueue.push({
-      task: () =>
-        writeJSON(path.resolve(this.projectPath, "swanky.config.json"), this.configBuilder, {
-          spaces: 2,
-        }),
+      task: () => {
+        if (Object.keys(this.swankyConfig.networks).length === 0 || this.swankyConfig.accounts.length === 0) {
+          this.swankyConfig = this.configBuilder as SwankyConfig;
+        } else {
+          this.swankyConfig.node = this.configBuilder.node!;
```

Comment:
We can improve type-safety and remove the usage of "!" for `configBuilder` by removing the `Partial<SwankyConfig>` utility typescript type. The `configBuilder` can be fully initialised with a dedicated `initSwankyConfig` helper used also in [swankyCommand.ts](src/lib/swankyCommand.ts). An example:

```typescript
configBuilder = initSwankyConfig();
configBuilder.defaultAccount = "alice";
```

Reply:
you are right no need to use `configBuilder` in init since we can use `this.swankyConfig`

Reply:
regarding `initSwankyConfig` maybe the name is a bit confusing.
what I had in mind was to use a builder pattern to construct the config something like this:

```typescript
export class SwankyConfigBuilder {
  private config: SwankyConfig;

  constructor(config: Partial<SwankyConfig>) {
    this.config = {
      node: {
        localPath: "",
        polkadotPalletVersions: swankyNode.polkadotPalletVersions,
        supportedInk: swankyNode.supportedInk,
      },
      accounts: [],
      networks: {
        local: { url: DEFAULT_NETWORK_URL },
        astar: { url: DEFAULT_ASTAR_NETWORK_URL },
        shiden: { url: DEFAULT_SHIDEN_NETWORK_URL },
        shibuya: { url: DEFAULT_SHIBUYA_NETWORK_URL },
      },
      contracts: {},
      ...config
    }
  }

  // then methods to update fields individually
  // e.g. updateAccounts
}
```

But this is must be done in a seperated issue, for now just do something like this:

```typescript
const buildSwankyConfig() => {
  node: {
      localPath: "",
      polkadotPalletVersions: swankyNode.polkadotPalletVersions,
      supportedInk: swankyNode.supportedInk,
    },
    accounts: [],
    networks: {
      local: { url: DEFAULT_NETWORK_URL },
      astar: { url: DEFAULT_ASTAR_NETWORK_URL },
      shiden: { url: DEFAULT_SHIDEN_NETWORK_URL },
      shibuya: { url: DEFAULT_SHIBUYA_NETWORK_URL },
    },
    contracts: {},
}
```